gateway/ln-gateway/src/actor.rs

```diff
@@ -147,43 +147,27 @@ impl GatewayActor {
         client: Arc<GatewayClient>,
         lnrpc: Arc<RwLock<dyn ILnRpcClient>>,
         route_hints: Vec<RouteHint>,
-        mut task_group: TaskGroup,
+        task_group: TaskGroup,
         gw_rpc: GatewayRpcSender,
     ) -> Result<Self> {
         let register_client = client.clone();
         let registration = register_client
             .config()
             .to_gateway_registration_info(route_hints.clone(), GW_ANNOUNCEMENT_TTL);
-        let registration_sent = registration.clone();
-        task_group
-            .spawn("Register with federation", |_| async move {
-                loop {
-                    // Retry gateway registration
-                    match retry(
-                        String::from("Register With Federation"),
-                        #[allow(clippy::unit_arg)]
-                        || async {
-                            Ok(register_client
-                                .register_with_federation(registration_sent.clone())
-                                .await?)
-                        },
-                        Duration::from_secs(1),
-                        5,
-                    )
-                    .await
-                    {
-                        Ok(_) => {
-                            info!("Connected with federation");
-                            tokio::time::sleep(GW_ANNOUNCEMENT_TTL / 2).await;
-                        }
-                        Err(e) => {
-                            warn!("Failed to connect with federation: {}", e);
-                            tokio::time::sleep(GW_ANNOUNCEMENT_TTL / 4).await;
-                        }
-                    }
-                }
```

Comment:
By removing the outer loop you remove re-registration. This means the gateway will disappear for clients after `GW_ANNOUNCEMENT_TTL` since the re-registration after `GW_ANNOUNCEMENT_TTL / 2` doesn't happen.

Maybe use `Notify` to wait for the first successful registration?